pkg/solver/store/db/db.go

```diff
@@ -0,0 +1,619 @@
+package store
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/lilypad-tech/lilypad/pkg/data"
+	"github.com/lilypad-tech/lilypad/pkg/solver/store"
+	"gorm.io/datatypes"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+	"gorm.io/gorm/logger"
+)
+
+type SolverStoreDatabase struct {
+	db *gorm.DB
+}
+
+func NewSolverStoreDatabase(connStr string, gormLogLevel string) (*SolverStoreDatabase, error) {
+	config := &gorm.Config{}
+	switch gormLogLevel {
+	case "silent":
+		config.Logger = logger.Default.LogMode(logger.Silent)
+	case "info":
+		config.Logger = logger.Default.LogMode(logger.Info)
+	case "error":
+		config.Logger = logger.Default.LogMode(logger.Error)
+	case "warn":
+		config.Logger = logger.Default.LogMode(logger.Warn)
+	default:
+		return nil, fmt.Errorf("expected solver store gorm log level silent, info, error, or warn, but received: %s", gormLogLevel)
```

Comment:
Can we pick a reasonable default here (info maybe?) so it doesn't need to be set (but can be overridden)?

Reply:
Gorm reports lookup errors in any mode other than `silent`. This is a bit noisy for looking up match decisions because we quite often will not have a match decision for a resource offer/job offer pairing.

Does `silent` seem like an acceptable default?

Reply:
`silent` works for me! Right, I remember that about gorm now. thanks!